core/api/metric.go

```diff
@@ -0,0 +1,18 @@
+package api
+
+import (
+	"github.com/intelsdi-x/snap/core"
+	"github.com/intelsdi-x/snap/core/serror"
+)
+
+type Metrics interface {
+	MetricCatalog() ([]core.CatalogedMetric, error)
+	FetchMetrics(core.Namespace, int) ([]core.CatalogedMetric, error)
+	GetMetricVersions(core.Namespace) ([]core.CatalogedMetric, error)
+	GetMetric(core.Namespace, int) (core.CatalogedMetric, error)
+	Load(*core.RequestedPlugin) (core.CatalogedPlugin, serror.SnapError)
+	Unload(core.Plugin) (core.CatalogedPlugin, serror.SnapError)
+	PluginCatalog() core.PluginCatalog
+	AvailablePlugins() []core.AvailablePlugin
+	GetAutodiscoverPaths() []string
+}
```

Comment:
I'll answer the 2 ways I understand your question:
1. We can't have a the API interface in `mgmt/rest` because `mgmt/rest/v2` (or v1) has a dependency on it ==> it would be a dependency loop
2. why having this package in `core/api` instead of `mgmt/rest/api`? I just thought it would make more sense because managers are implementing interfaces from `core`.